src/Avalonia.Visuals/Media/PathFigure.cs

```diff
@@ -33,6 +42,30 @@ public PathFigure()
             Segments = new PathSegments();
         }
 
+        static PathFigure()
+        {
+            SegmentsProperty.Changed.AddClassHandler<PathFigure>((s, e) =>
+                s.OnSegmentsChanged(e.NewValue as PathSegments));
+        }
+
+        private void OnSegmentsChanged(PathSegments? arg2NewValue)
+        {
+            _segmentsObserver?.Dispose();
+            _segmentsPropertiesObserver?.Dispose();
+
+            _segmentsObserver = _segments?.ForEachItem(
```

Comment:
Nit: this isn't an observable. I'd call it `_segmentsSubscription` or something.